### PR TITLE
metric: Update doc with info on rules

### PR DESCRIPTION
**Pull request by rimadeodhar**
This PR updates doc.go within the metric package
to provide information on how the new rules
framework should be used and tested. The information
includes pointers on when to use an AlertingRule vs.
an AggregationRule as well as how to test it.

Release note: None

### CODE REVIEW

**Review by dhartunian — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73571)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian and @rimadeodhar)

___
*[pkg/util/metric/doc.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZhYZ8OcWT-n2MvPn:-MqPZhYZ8OcWT-n2MvPo:b3x3uua) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L20)):*
> ```Go
> format at /api/v2/rules/. These rules are not consumed programmatically and are
> meant to be used as guidelines from engineers implementing metric(s) on how
> metric(s) can be aggregated and used in alerts.
> ```

Maybe reword or break up this sentence. Don't we expect some usage of this w/ prometheus?
___
*[pkg/util/metric/doc.go, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZqxy1LqGJfM_iLZG:-MqPZqxy1LqGJfM_iLZH:bcqz0ym) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L69)):*
> ```Go
> There are two types of rules:
> 1. AlertingRule: This rule is used to provide guidelines on how
>    one or metric can be used for alerts.
> ```

or metric -> or more metrics
___
*[pkg/util/metric/doc.go, line 71 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZyAU0YkfPRKb9FyW:-MqPZyAU0YkfPRKb9FyX:b-jop0pu) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L71)):*
> ```Go
>    one or metric can be used for alerts.
> 2. AggregationRule: This rule is used to provide guidelines on
>    how one or more metric can be aggregated to provide indicators
> ```

metrics
___
*[pkg/util/metric/doc.go, line 72 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZza25Ud7oiH-kRi9:-MqPZza25Ud7oiH-kRiA:b-ampbfo) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L72)):*
> ```Go
> 2. AggregationRule: This rule is used to provide guidelines on
>    how one or more metric can be aggregated to provide indicators
>    about the system health.
> ```

remove "the"
___
*[pkg/util/metric/doc.go, line 118 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqP_HLN1Y5Zwm9UHdEA:-MqP_HLN1Y5Zwm9UHdEB:b96i0zn) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L118)):*
> ```Go
> 
> To ensure the rules you added work as expected, you can use the GetRulesForTest helper API within
> the RuleRegistry to confirm the rules are initialized without any error.
> ```

unrelated to this PR: do we have the ability to just load all the rules into the registry in a test and make sure the endpoint returns a valid response? maybe we already do this...

**Review by rimadeodhar — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73571)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/util/metric/doc.go, line 20 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZhYZ8OcWT-n2MvPn:-MqQHhh_EdU7xr4rqwER:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L20)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

Maybe reword or break up this sentence. Don't we expect some usage of this w/ prometheus?
</blockquote></details>

Done.
___
*[pkg/util/metric/doc.go, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZqxy1LqGJfM_iLZG:-MqQHhBD4dnl-SD8cTcX:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L69)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

or metric -> or more metrics
</blockquote></details>

Done.
___
*[pkg/util/metric/doc.go, line 71 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZyAU0YkfPRKb9FyW:-MqQHgjk1-aUDgu264uT:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L71)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

metrics
</blockquote></details>

Done.
___
*[pkg/util/metric/doc.go, line 72 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZza25Ud7oiH-kRi9:-MqQHgUX989U7BhB9xEs:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L72)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

remove "the"
</blockquote></details>

Done.
___
*[pkg/util/metric/doc.go, line 118 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqP_HLN1Y5Zwm9UHdEA:-MqQHYBq9Q2-6xBKFJia:bfcwpbq) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L118)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

unrelated to this PR: do we have the ability to just load all the rules into the registry in a test and make sure the endpoint returns a valid response? maybe we already do this...
</blockquote></details>

Yes, we have an integration test for the api/v2 endpoint which will verify that all rules added are valid.

**Review by abarganier — APPROVED** [review]
Looks good, I think we missed one piece of David's feedback, but :lgtm: once that's fixed :thumbsup: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-:-MrTLqCb574MYizFSciy:b-af79ln)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dhartunian)

___
*[pkg/util/metric/doc.go, line 69 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/73571#-MqPZqxy1LqGJfM_iLZG:-MrTLCa4BgoWwM2i4tat:b5rxa02) ([raw file](https://github.com/cockroachdb/cockroach/blob/24be728ede03bfe2f961080c1feb8ee4fb7986e1/pkg/util/metric/doc.go#L69)):*
<details><summary><i>Previously, rimadeodhar (Rima Deodhar) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

nit: I think this was missed with the latest patch :)

**Comment by rimadeodhar** [comment]
TFTR!

bors r=abarganier
